addons/purchase/tests/test_purchase.py

```diff
@@ -501,3 +501,36 @@ def test_supplier_list_in_product_with_multicompany(self):
         product._invalidate_cache()
         self.assertEqual(product.seller_ids[0].partner_id, self.partner_a)
         self.assertEqual(product.seller_ids[0].company_id, company_a)
+
+    def test_purchase_product_search(self):
+        """ This test ensures that the product can be found by its name, not only vendor name
+        """
+
+        #create a contact of type contact
+        contact = self.env['res.partner'].create({
+            'name': 'Contact',
+            'type': 'contact',
+        })
+
+        #create a contact of type Delivery Address lnked to the contact
+        delivery_address = self.env['res.partner'].create({
+            'name': 'Delivery Address',
+            'type': 'delivery',
+            'parent_id': contact.id,
+        })
```

Comment:
Why not using only the contact instead of this delivery address? IMO you could just as well use a partner your `ref` from demo data instead of having to create one.